lib/polymorphic_integer_type/extensions.rb

```diff
@@ -24,26 +24,32 @@ def belongs_to(name, scope = nil, **options)
             mapping
           end
 
-          define_method foreign_type do
-            t = super()
-            mapping[t]
-          end
+          component_extension = Module.new do
+            define_method foreign_type do
+              t = super()
+              self.class.send("#{foreign_type}_mapping")[t]
```

Comment:
`mapping` is a a local variable in the context of the `belongs_to` method. It was captured in the context of the dynamic methods. We have the ability to access the mapping through a class variable, so we can use that method instead of capturing the local variable.